src/rgw/rgw_bucket.cc

```diff
@@ -2968,20 +2989,23 @@ int RGWBucketCtl::do_link_bucket(RGWSI_Bucket_EP_Ctx& ctx,
       ep = pinfo->ep;
       pattrs = &pinfo->attrs;
     } else {
+      std::unique_ptr<jaeger_tracing::Span> span_1 = jaeger_tracing::child_span("svc_bucket_sobj.cc : RGWSI_Bucket_SObj::read_bucket_entrypoint_info", span.get());
       ret = svc.bucket->read_bucket_entrypoint_info(ctx,
                                                     meta_key,
                                                     &ep, &rot,
                                                     nullptr, &attrs,
                                                     y);
+      jaeger_tracing::finish_trace(span_1.get());
       if (ret < 0 && ret != -ENOENT) {
         ldout(cct, 0) << "ERROR: store->get_bucket_entrypoint_info() returned: "
                       << cpp_strerror(-ret) << dendl;
       }
       pattrs = &attrs;
     }
   }
-
+  std::unique_ptr<jaeger_tracing::Span> span_2 = jaeger_tracing::child_span("rgw_user.cc : RGWUserCtl::add_bucket", span.get());
```

Comment:
[nit] you can use `auto` or maybe `const auto` here and in all other places where you create a new span or child span

Reply:
noted